baybe/searchspace/continuous.py

```diff
@@ -247,6 +272,15 @@ def param_names(self) -> tuple[str, ...]:
         """Return list of parameter names."""
         return tuple(p.name for p in self.parameters)
 
+    @property
+    def param_names_in_cardinality_constraint(self) -> tuple[str, ...]:
+        """Return list of parameter names involved in cardinality constraints."""
+        # TODO: Is this property really needed? If so, apply naming conventions.
+        params_per_cardinatliy_constraint = [
+            c.parameters for c in self.constraints_cardinality
+        ]
+        return tuple(chain(*params_per_cardinatliy_constraint))
```

Comment:
No, at the moment it can't, because our entire machinery around cardinality constraints is built onto the assumption that the constraints are non-overlapping in their parameters, which is ensured right in the member validator. That said, this core assumption is baked in to the logic of most other functions as well. Therefore, if we ever gonna loosen the assumption, we need to revisit all downstream functions anyway. So I'd skip the set operation for now. But let me know if you have concerns.